dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+import com.google.cloud.dlp.v2.DlpServiceClient;
+import com.google.privacy.dlp.v2.ByteContentItem;
+import com.google.privacy.dlp.v2.ContentItem;
+import com.google.privacy.dlp.v2.CustomInfoType;
+import com.google.privacy.dlp.v2.Finding;
+import com.google.privacy.dlp.v2.InfoType;
+import com.google.privacy.dlp.v2.InspectConfig;
+import com.google.privacy.dlp.v2.InspectContentRequest;
+import com.google.privacy.dlp.v2.InspectContentResponse;
+import com.google.privacy.dlp.v2.LocationName;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+
+// [START dlp_inspect_augment_infotypes]
+
+
+public class InspectStringAugmentInfoType {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // The string to de-identify.
+    String textToInspect = "patient: quasimodo";
+    inspectStringAugmentInfoType(projectId, textToInspect);
+  }
+
+  // Inspects the provided text.
+  public static void inspectStringAugmentInfoType(String projectId, String textToInspect)
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DlpServiceClient dlp = DlpServiceClient.create()) {
+      // Specify the type and content to be inspected.
+      ByteContentItem byteItem =
+          ByteContentItem.newBuilder()
+              .setType(ByteContentItem.BytesType.TEXT_UTF8)
+              .setData(ByteString.copyFromUtf8(textToInspect))
+              .build();
+      ContentItem item = ContentItem.newBuilder().setByteItem(byteItem).build();
+
+      // Construct the custom word list to be detected.
+      CustomInfoType.Dictionary wordList =
+          CustomInfoType.Dictionary.newBuilder()
+              .setWordList(
+                  CustomInfoType.Dictionary.WordList.newBuilder().addWords("quasimodo").build())
+              .build();
+
+      InfoType infoType = InfoType.newBuilder().setName("PERSON_NAME").build();
+      // Construct the custom infotype detector associated with the word list.
```

Comment:
// Construct a custom infotype detector by augmenting the PERSON_NAME detector with a word list